ibis/tests/all/conftest.py

```diff
@@ -1,63 +1,62 @@
+import os
 import pytest
 
-import ibis
+from ibis.compat import Path
+from ibis.tests.backends import (Csv, Parquet, Pandas,
+                                 SQLite, Postgres, MySQL,
+                                 Clickhouse, Impala, BigQuery)
 
-from ibis.tests.all import backends
 
-
-@pytest.fixture(
-    params=backends,
-    scope='session',
-    ids=lambda cls: cls.__name__.lower(),
-)
-def backend(request):
-    return request.param
+pytestmark = pytest.mark.backend
 
 
 @pytest.fixture(scope='session')
-def con(backend):
-    backend_module_name = backend.__name__.lower()
-    try:
-        module = getattr(ibis, backend_module_name)
-    except AttributeError as e:
-        pytest.skip(
-            'Unable to import the {} backend: {}'.format(
-                backend_module_name, e
-            )
-        )
-    return backend.connect(module)
+def data_directory():
+    root = Path(__file__).absolute().parents[3]
 
+    default = root / 'testing' / 'ibis-testing-data'
+    datadir = os.environ.get('IBIS_TEST_DATA_DIRECTORY', default)
+    datadir = Path(datadir)
 
-@pytest.fixture(scope='session')
-def dialect(con):
-    return con.dialect
+    if not datadir.exists():
+        pytest.skip('test data directory not found')
 
+    return datadir
 
-@pytest.fixture(scope='session')
-def translator(dialect):
-    return dialect.translator
+
+@pytest.fixture(params=[
+    pytest.param(Csv, marks=pytest.mark.csv),
+    pytest.param(Parquet, marks=pytest.mark.parquet),
+    pytest.param(Pandas, marks=pytest.mark.pandas),
+    pytest.param(SQLite, marks=pytest.mark.sqlite),
+    pytest.param(Postgres, marks=pytest.mark.postgres),
+    pytest.param(MySQL, marks=pytest.mark.mysql),
+    pytest.param(Clickhouse, marks=pytest.mark.clickhouse),
+    pytest.param(BigQuery, marks=pytest.mark.bigquery),
+    pytest.param(Impala, marks=pytest.mark.impala)
```

Comment:
Ok, that is awesome. I was wondering if there was a sane way to mark parameterized fixtures.

Reply:
I've almost written a pytest plugin :)